tests/components/rest/test_sensor.py

```diff
@@ -559,6 +559,39 @@ def test_update_with_xml_convert_json_attrs_with_jsonattr_template(self):
         assert "12556" == self.sensor.device_state_attributes["ver"]
         assert "bogus" == self.sensor.state
 
+    def test_update_with_application_xml_convert_json_attrs_with_jsonattr_template(
+        self,
+    ):
+        """Test attributes get extracted from a JSON result that was converted from XML with application/xml mime type."""
+        json_attrs_path = "$.main"
+        value_template = template("{{ value_json.main.dog }}")
+        value_template.hass = self.hass
+
+        self.rest.update = Mock(
+            "rest.RestData.update",
+            side_effect=self.update_side_effect(
+                "<main><dog>1</dog><cat>3</cat></main>",
+                CaseInsensitiveDict({"Content-Type": "application/xml"}),
+            ),
+        )
+        self.sensor = rest.RestSensor(
+            self.hass,
+            self.rest,
+            self.name,
+            self.unit_of_measurement,
+            self.device_class,
+            value_template,
+            ["dog", "cat"],
+            self.force_update,
+            self.resource_template,
+            json_attrs_path,
+        )
+
+        self.sensor.update()
+        assert "3" == self.sensor.device_state_attributes["cat"]
+        assert "1" == self.sensor.device_state_attributes["dog"]
+        assert "1" == self.sensor.state
```

Comment:
Is it meaningful to assert the same value twice? Or rather does it make sense to add `dog` as a json_attrs if it is already defined as the state of the sensor.

Reply:
@springstan  Its testing both the attribute parsing path and the template lookup even if they point to the same place it still provides coverage of both paths.

Reply:
Yeah you are correct, I was just wondering why you used the same place for both in comparison to other tests.

Reply:
Since there wasn't one that did it that way it gave more regression surface in case someone in the future introduces something that interacts poorly between the template and jsonpath.  Seems unlikely but you never know.